packages/builder/src/builderStore/store/frontend.js

```diff
@@ -1040,6 +1042,30 @@ export const getFrontendStore = () => {
           if (component[name] === value) {
             return false
           }
+
+          const settings = getComponentSettings(component._component)
+          const updatedSetting = settings.find(setting => setting.key === name)
+
+          if (
+            updatedSetting.type === "dataSource" ||
+            updatedSetting.type === "table"
+          ) {
+            const { list: allTables } = get(tables)
+            const { schema } = allTables.find(
+              ({ _id }) => _id === value.tableId
```

Comment:
It's possible to set a `table` type setting back to `undefined` - so we'll need optional chaining here on `value.tableId`.